src/base/server_type.cpp

```diff
@@ -223,6 +225,8 @@ int setup_tran_server_params_on_ha_mode ()
 
   int port_id = prm_get_master_port_id ();
 
+  const char *localhost_str = "localhost";
```

Comment:
something with `constexpr`;
maybe even hardcode the string where needed

Reply:
OK, done.